Compile and install without root.

rot13 opened the pull request:
Summary of changes:
- move apt-get invocations to a separate script
- set proper install prefix in:
  - cmake invocations
  - for angelscript and mysocketw - makefile variables
  - for RoR itself - plugins.cfg file
- move common variables to a separate config file

Unrelated changes:
- add lazy man's error handling using "set -eu"
- handle existing files and directories (add -c to wget, -o to zip, run git clone conditionally)
- add git pull to update existing repositories

rot13 committed:
Add separate script for dependencies installed by apt-get.

rot13 committed:
Rework scripts to compile and install everything without root.

Paths are specified in a separate config file.
By default sources are downloaded to ~/rigs-of-rods/source,
and binaries are installed to ~/rigs-of-rods/install.

Added "set -eu" to stop on errors or undefined variables.
Added "-c" to wget, to continue interrupted transfers.
Added "-o" to unzip, so it doesn't stop if files are already there.
Added "git pull" to update git repositories when the script is run again.

rot13 committed:
Remove RoRupdate.sh script, as it is no longer needed.

rot13 committed:
Update documentation.

rot13 committed:
Remove the note from README as it no longer applies.

rot13 committed:
Making scripts executable is not required.

rot13 committed:
Added OpenAL to fix Agora bus crash.

Thanks to Bob-Z for reporting the problem and finding a solution.

rot13 committed:
Use absolute path in PluginFolder config option.

rot13 committed:
Add forgotten --depth=1 to openal git clone.

rot13 committed:
Add comment to RoRcore why we use cp instead of make install.

rot13 committed:
Add comments to config options.

rot13 committed:
Add some notes about the config file to README.

Hiradur commented:
Great improvements. However, in a VM running a fresh install of Xubuntu 14.04, CMake can't find OGRE. Any ideas?

I will probably comment out OpenAL because it's broken every now and then in upstream.

rot13 committed:
Export PKG_CONFIG_PATH variable.

Testing revealed cmake won't find packages on Ubuntu 14.04 if this variable is not exported.

rot13 committed:
Add wget to system dependencies.

rot13 committed:
Switch to version 1.16.0 of openal.

rot13 commented:
Thank you for testing. I tried it in 14.04 chroot created by debootstrap, and it seems exporting PKG_CONFIG_PATH fixes the problem. It was also missing wget, so I added it, and switched to a stable release of openal - it still fixes the crash for me, so I guess using bleeding edge git master is not required.

Hiradur commented:
That fixed it. Thank you very much for your contribution.
